core/domain/user_jobs_test.py

```diff
@@ -651,3 +794,105 @@ def test_deleted_exploration(self):
+            self.save_new_valid_collection(
+                self.COLLECTION_ID, self.user_a_id,
+                exploration_id=self.COLLECTION_EXP_ID)
+
+            # User A adds user B as an editor to the collection.
+            rights_manager.assign_role_for_collection(
+                self.user_a_id, self.COLLECTION_ID, self.user_b_id,
+                rights_manager.ROLE_EDITOR)
+            # User A adds user C as a viewer of the collection.
+            rights_manager.assign_role_for_collection(
+                self.user_a_id, self.COLLECTION_ID, self.user_c_id,
+                rights_manager.ROLE_VIEWER)
+
+        self._run_one_off_job()
+
+        # Users A and B are subscribed to the collection. User C is not.
+        user_a_subscriptions_model = user_models.UserSubscriptionsModel.get(
+            self.user_a_id)
+        user_b_subscriptions_model = user_models.UserSubscriptionsModel.get(
+            self.user_b_id)
+        user_c_subscriptions_model = user_models.UserSubscriptionsModel.get(
+            self.user_c_id, strict=False)
+
+        self.assertEqual(
+            user_a_subscriptions_model.collection_ids, [self.COLLECTION_ID])
+        # User A is also subscribed to the exploration within the collection
+        # because they created both.
+        self.assertEqual(
+            sorted(user_a_subscriptions_model.activity_ids), [
+                self.COLLECTION_EXP_ID, self.EXP_ID])
+        self.assertEqual(
+            user_b_subscriptions_model.collection_ids, [self.COLLECTION_ID])
+        self.assertEqual(user_a_subscriptions_model.feedback_thread_ids, [])
+        self.assertEqual(user_b_subscriptions_model.feedback_thread_ids, [])
+        self.assertEqual(user_c_subscriptions_model, None)
+
+    def test_two_collections(self):
+        with self.swap(
+                subscription_services, 'subscribe_to_thread', self._null_fn
+            ), self.swap(
+                subscription_services, 'subscribe_to_exploration',
+                self._null_fn
+            ), self.swap(
+                subscription_services, 'subscribe_to_collection',
+                self._null_fn):
+            # User A creates and saves a new valid collection.
+            self.save_new_valid_collection(
+                self.COLLECTION_ID, self.user_a_id,
+                exploration_id=self.COLLECTION_EXP_ID)
+
+            # User A creates and saves another valid collection.
+            self.save_new_valid_collection(
+                self.COLLECTION_ID_2, self.user_a_id,
+                exploration_id=self.COLLECTION_EXP_ID)
+
+        self._run_one_off_job()
+
+        # User A is subscribed to two collections.
+        user_a_subscriptions_model = user_models.UserSubscriptionsModel.get(
+            self.user_a_id)
+
+        self.assertEqual(
+            sorted(user_a_subscriptions_model.collection_ids),
+            sorted([self.COLLECTION_ID, self.COLLECTION_ID_2]))
+
+    def test_deleted_collection(self):
```

Comment:
I'm not so sure this needs to be tested. The test is testing against an empty collection.

Any behavior with subscribing to explorations happens as part of the standard exploration subscriptions pipeline and is guarded by the exploration subscription tests. Collections are independent of this behavior and have been proved in the immediate pipeline not to lead to collections and explorations from different authors crossing subscription. I added one test to guard against similar behavior in the user job.